tool/tctl/common/resource_command.go

```diff
@@ -616,6 +623,24 @@ func (rc *ResourceCommand) Delete(client auth.ClientI) (err error) {
 			return trace.Wrap(err)
 		}
 		fmt.Printf("lock %q has been deleted\n", rc.ref.Name)
+	case types.KindDatabaseServer:
+		dbServers, err := client.GetDatabaseServers(ctx, apidefaults.Namespace)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		deleted := false
+		for _, server := range dbServers {
+			if server.GetName() == rc.ref.Name {
+				if err := client.DeleteDatabaseServer(ctx, apidefaults.Namespace, server.GetHostID(), server.GetName()); err != nil {
+					return trace.Wrap(err)
+				}
+				deleted = true
+			}
+		}
+		if !deleted {
+			return trace.NotFound("database server %q not found", rc.ref.Name)
+		}
```

Comment:
Do we return errors on other non-existent resources? We should make behavior consistent.

Reply:
Yes, For example: https://github.com/gravitational/teleport/blob/650b840fa381cbac9d4e15d3976a929c768b1d81/tool/tctl/common/resource_command.go#L960

Reply:
@smallinsky The example you gave is for getting the resource though, not deleting it, right? When deleting, do we error on non-existent kube service for example? The case above just does `DeleteKubeService` and doesn't iterate over them.

Reply:
> The example you gave is for getting the resource though, not deleting it, right?

right , but for example when `tctl rm command` is called for not existing users resource:
 
`tctl rm users/invalidusername`

the `NotFound` error is returned: 
```
ERROR REPORT:
Original Error: *trace.NotFoundError user &#34;invalidusername&#34; is not found
Stack Trace:
	/Users/marek/Go/src/github.com/gravitational/teleport/vendor/github.com/gravitational/teleport/api/client/client.go:515 github.com/gravitational/teleport/api/client.(*Client).DeleteUser
	/Users/marek/Go/src/github.com/gravitational/teleport/tool/tctl/common/resource_command.go:557 github.com/gravitational/teleport/tool/tctl/common.(*ResourceCommand).Delete
	/Users/marek/Go/src/github.com/gravitational/teleport/tool/tctl/common/resource_command.go:146 github.com/gravitational/teleport/tool/tctl/common.(*ResourceCommand).TryRun
	/Users/marek/Go/src/github.com/gravitational/teleport/tool/tctl/common/tctl.go:167 github.com/gravitational/teleport/tool/tctl/common.Run
	/Users/marek/Go/src/github.com/gravitational/teleport/tool/tctl/main.go:36 main.main
	/usr/local/Cellar/go/1.16.3/libexec/src/runtime/proc.go:225 runtime.main
	/usr/local/Cellar/go/1.16.3/libexec/src/runtime/asm_amd64.s:1371 runtime.goexit
User Message: user &#34;invalidusername&#34; is not found
```


Does